src/glyph/parse.rs

```diff
@@ -46,43 +50,91 @@ impl<'names> GlifParser<'names> {
         raw_xml: &[u8],
         buf: &mut Vec<u8>,
     ) -> Result<Glyph, GlifLoadError> {
+        let mut seen_advance = false;
+        let mut seen_image = false;
+        let mut seen_lib = false;
+        let mut seen_note = false;
+        let mut seen_outline = false;
```

Comment:
can't we just check the various parts of the glyph object to see if they've been set?

Reply:
Well, outline, lib and advance.(height|width) aren't using Option, so we can't tell apart not-yet-encountered and default-value-of-zero-or-empty... but yes for image and note.